src/solvers/petsc_nonlinear_solver.C

```diff
@@ -782,13 +778,22 @@ PetscNonlinearSolver<T>::solve (SparseMatrix<T> &  pre_in,  // System Preconditi
             ierr = MatSetNearNullSpace(J, msp);
             LIBMESH_CHKERR(ierr);
           }
-
           ierr = MatNullSpaceDestroy(&msp);
           LIBMESH_CHKERR(ierr);
         }
     }
 #endif
 
+if (_solve_type != STANDARD)
+  {
+#if PETSC_VERSION_LESS_THAN(3,2,0)
+    ierr = MatDestroy(J);
+#else
+    ierr = MatDestroy(&J);
```

Comment:
For this, please use the `LibMeshMatDestroy()` macro which is already set up to handle the API change.